lib/Containers/MerkleTree.cpp

```diff
@@ -1045,13 +1340,16 @@ void MerkleTree<Hasher, BranchingBits>::leftCombine(bool withShift) noexcept {
     // ...
     // empty -> n
     for (std::uint64_t i = 2; i < n; ++i) {
-      Node& src = this->node(i);
-      Node& dst = this->node((i + 1) / 2);
-
-      TRI_ASSERT(&src != &dst);
-      dst.count += src.count;
-      dst.hash ^= src.hash;
-      src = Node{0, 0};
+      if (!this->empty(i)) {
```

Comment:
I fear this change is not valid. For example, we combine buckets 3 and 4 into 2. If 3 and 4 are both empty (not even allocated), then 2 must be empty after the transformation. However, this if in line 1343 would prevent any change in bucket 2, so that if there was data in bucket 2 before, it would now be wrong!
I fear we must change the loop to do one combination in one iteration and distinguish 4 cases:
  (1) all empty ==> do nothing
  (2) destination empty, at least one of the sources not ==> copy and combine
  (3) destination non-empty, at least one of the sources not ==> copy and combine
  (4) destination non-empty, both sources empty ==> make destination empty
And: Bucket 0 and buckets 1,2 -> 1 are special, because of aliasing!

Reply:
This comment was wrong. Due to the fact that earlier loop iterations had the current `dst` as a `src` and have then erased it, all is well.

Reply:
I have added a comment to explain this in the code.